src/lints/trait_removed_associated_type.ron

```diff
@@ -0,0 +1,54 @@
+SemverQuery(
+    id: "trait_removed_associated_type",
+    human_readable_name: "associated type of trait removed",
+    description: "A trait's associated type was removed.",
+    required_update: Major,
+    reference_link: None,
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Trait {
+                        trait_name: name @output
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        associated_type {
+                            associated_type: name @output @tag
+
+                            span_: span {
+                                filename @output
+                                begin_line @output
+                            }
+                        }
+                    }
+                }
+            }
+            current @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+
+                        associated_type {
+                            name @filter(op: "=", value: ["%associated_type"])
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "zero": 0,
+    },
+    error_message: "A public trait's associated type was removed.",
+    per_result_error_template: Some("trait {{trait_name}} no longer has a type {{associated_type}}, previously at {{span_filename}}:{{span_begin_line}}"),
```

Comment:
Do we want to output the file path it used to be at? Not super useful if they already know what the trait name is and what the removed type is.

Reply:
Yes, I think we do. Most often when this lint fires, the trait is still going to be in the same file and more-or-less around the same line number. This would be useful to the user so they don't have to hunt for the trait themselves, and also matches the behavior of other lints.

Also, slight wording tweak suggestion:
```suggestion
    per_result_error_template: Some("trait {{trait_name}} no longer has associated type {{associated_type}}, previously at {{span_filename}}:{{span_begin_line}}"),
```

Reply:
Sounds good

Reply:
Oh actually, another potentially better phrasing:
```
trait associated type {{trait_name}}::{{associated_type}}, previously at {{span_filename}}:{{span_begin_line}}
```

This avoids the repetition from the error message and is more concise while saying the same thing.

Reply:
[a583932](https://github.com/obi1kenobi/cargo-semver-checks/pull/500/commits/a583932b6e379969c74c1608a1a2cd021a5c23da)